src/libs/actions/Report.js

```diff
@@ -328,15 +331,24 @@ function fetchChatReportsByIDs(chatList) {
  *
  * @param {Number} reportID
  * @param {Number} sequenceNumber
+ * @param {Boolean} saveNewMarkerPosition
  */
-function setLocalLastRead(reportID, sequenceNumber) {
+function setLocalLastRead(reportID, sequenceNumber, saveNewMarkerPosition) {
     lastReadSequenceNumbers[reportID] = sequenceNumber;
 
     // Update the report optimistically
-    Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
-        unreadActionCount: 0,
-        lastVisitedTimestamp: Date.now(),
-    });
+    if (saveNewMarkerPosition) {
+        Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
+            unreadActionCount: Math.max(reportMaxSequenceNumbers[reportID] - sequenceNumber, 0),
+            lastVisitedTimestamp: Date.now(),
+            newMarkerPosition: saveNewMarkerPosition ? sequenceNumber : 0,
+        });
+    } else {
+        Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT}${reportID}`, {
+            unreadActionCount: 0,
```

Comment:
I'm not sure I understand the situation where we will won't want to calculate the `unreadActionCount` exactly as we are doing if the first block of this `if/else` ?

And if that's true then we really don't need this `if/else`. Just update the local last read the same every time. Then if you need to set the `newMarkerPosition` do it right after calling `setLocalLastRead()` ?

Reply:
If we update the unreadActionCount to the calculated value (instead of zero) in the previously existing flow (so, when receiving a new message), the chat in the LHB gets highlighted whenever the user sends a message, so we need to set it to zero like we used to.

Reply:
>in the previously existing flow (so, when receiving a new message), the chat in the LHB gets highlighted whenever the user sends a message

Sorry, still not sure I understand the situation you are describing. From looking at what this method did before we only called this when:

1. The user marked the report as read via `updateLastReadActionID`
2. We handle a report comment from the currently logged in user

So I'm curious if it makes sense to still have that as a "default behavior" given the changes we are now making.

Reply:
Well, those are the two cases we are handling. In case 2, we need to set it to zero (otherwise, the chat get highlighted in the LHN).
We can certainly rename the flag to something like "isNewSequenceFromCurrentUser" or something like that, or maybe split this in two functions, one for each case?